mm2src/mm2_db/src/indexed_db/drivers/cursor/cursor.rs

```diff
@@ -311,25 +371,52 @@ impl CursorDriver {
         let (item_action, cursor_action) = self.inner.on_iteration(key)?;
 
         let (id, val) = item.into_pair();
+
         // Checks if the given `where_` condition, represented by an optional closure (`cursor_condition`),
         // is satisfied for the provided `item`. If the condition is met, return the corresponding `(id, val)` or skip to the next item.
-
         if matches!(item_action, CursorItemAction::Include) {
-            if let Some(cursor_condition) = where_ {
+            if let Some(cursor_condition) = &self.filters_ext.where_ {
                 if cursor_condition(val.clone())? {
-                    // stop iteration and return value.
-                    self.stopped = true;
+                    // Update limit (if applicable) and return
+                    if self.filters_ext.limit.is_some() {
+                        self.update_limit_and_continue(&cursor, &cursor_action).await?;
+                    } else {
+                        self.stopped = true;
+                    };
                     return Ok(Some((id, val)));
```

Comment:
So if limit is not specified we just return first found value, right?

Reply:
right.